File: Data Structures & Algorithms/duplicate-integer/submission-1.java
class Solution {
    public boolean hasDuplicate(int[] nums) {
        // An *number* : *is present* map
        Map<Integer, Boolean> map = new HashMap<>();

        for (int num : nums) {
            if (map.containsKey(num)) return true;
            map.put(num, true);
        }

        return false;
    }
}

File: Data Structures & Algorithms/find-duplicate-integer/submission-1.ts
class Solution {
    /**
     * @param {number[]} nums
     * @return {number}
     */
    findDuplicate(nums: number[]): number {
        let seen: Set<number> = new Set<number>();

        for (const num of nums) {
            if (seen.has(num)) {
                return num;
            }

            seen.add(num);
        }

        return -1;
    }
}
